lib/charms/identity_platform_login_ui_operator/v0/login_ui_endpoints.py

```diff
@@ -0,0 +1,179 @@
+)
+Class SomeCharm(CharmBase):
+    def __init__(self, *args):
+        self.login_ui_endpoints_relation = LoginUIEndpointsRequirer(self)
+        self.framework.observe(self.on.some_event_emitted, self.some_event_function)
+    def some_event_function():
+        # fetch the relation info
+        try:
+            login_ui_endpoints = self.login_ui_endpoints_relation.get_login_ui_endpoints()
+        except LoginUIEndpointsRelationError as error:
+            ...
+```
+"""
+
+import logging
+from typing import Dict, Optional
+
+from ops.charm import CharmBase, RelationCreatedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import Application
+
+# The unique Charmhub library identifier, never change it
+LIBID = "460ab09e6b874d1c891b67f83586c9a7"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 2
+
+RELATION_NAME = "ui-endpoint-info"
+INTERFACE_NAME = "login_ui_endpoints"
+logger = logging.getLogger(__name__)
+
+RELATION_KEYS = [
+    "consent",
+    "error",
+    "index",
+    "login",
+    "oidc_error",
+    "registration",
+    "browser",
+]
+
+
+class LoginUIEndpointsRelationReadyEvent(EventBase):
+    """Event to notify the charm that the relation is ready."""
+
+
+class LoginUIEndpointsProviderEvents(ObjectEvents):
+    """Event descriptor for events raised by `LoginUIEndpointsProvider`."""
+
+    ready = EventSource(LoginUIEndpointsRelationReadyEvent)
+
+
+class LoginUIEndpointsProvider(Object):
+    """Provider side of the endpoint-info relation."""
+
+    on = LoginUIEndpointsProviderEvents()
+
+    def __init__(self, charm: CharmBase, relation_name: str = RELATION_NAME):
+        super().__init__(charm, relation_name)
+
+        self._charm = charm
+        self._relation_name = relation_name
+
+        events = self._charm.on[relation_name]
+        self.framework.observe(
+            events.relation_created, self._on_provider_endpoints_relation_created
+        )
+
+    def _on_provider_endpoints_relation_created(self, event: RelationCreatedEvent) -> None:
+        self.on.ready.emit()
+
+    def send_endpoints_relation_data(self, endpoint: str) -> None:
+        """Updates relation with endpoint info."""
+        if not self._charm.unit.is_leader():
+            return
+
+        relations = self.model.relations[self._relation_name]
+        for relation in relations:
+            relation.data[self._charm.app].update(
+                {
+                    "consent": f"{endpoint}/consent",
+                    "error": f"{endpoint}/error",
+                    "index": f"{endpoint}/index",
+                    "login": f"{endpoint}/login",
+                    "oidc_error": f"{endpoint}/oidc_error",
+                    "registration": f"{endpoint}/registration",
+                    "browser": endpoint,
```

Comment:
`browser` is not a very descriptive name, perhaps something like `index_url` or `default_url` would be better?

Reply:
changed to default_url